osu.Game.Rulesets.Osu/Statistics/AccuracyHeatmap.cs

```diff
@@ -227,9 +232,9 @@ private void load()
             }
         }
 
-        protected void AddPoint(Vector2 start, Vector2 end, Vector2 hitPoint, float radius)
+        public void AddPoint(Vector2 start, Vector2 end, Vector2 hitPoint, float radius)
         {
-            if (pointGrid.Content.Count == 0)
+            if (pointGrid == null || pointGrid.Content.Count == 0)
```

Comment:
what is with the null check here? sounds to me like you're calling `AddPoint()` too early before the `pointGrid` has been created and trying to paper over that crack by doing this. this is not the correct way to do this, you must somehow make sure that the calls to `AddPoint()` that occur too early take place when everything else is ready. many ways to do this, including but not limited to scheduling or queueing the calls in a list until loaded